ignite/pkg/protoanalysis/protoutil/helpers.go

```diff
@@ -248,6 +248,39 @@ func GetImportByPath(f *proto.Proto, path string) (node *proto.Import, err error
 	return nil, errors.Errorf("import %s not found", path)
 }
 
+// GetFieldByName returns the field with the given name or nil if not found within a message.
+// Only traverses in proto.Message since they are the only nodes that contain fields:
+//
+//		f, _ := ParseProtoPath("foo.proto")
+//		m := GetMessageByName(f, "Foo")
+//		f := GetFieldByName(m, "Bar")
+//	 f.Name // "Bar"
+func GetFieldByName(f *proto.Message, name string) (node *proto.NormalField, err error) {
+	node, err = nil, nil
+	found := false
+	Apply(f,
+		func(c *Cursor) bool {
+			if m, ok := c.Node().(*proto.NormalField); ok {
+				if m.Name == name {
+					found = true
+					node = m
+					return false
+				}
+				// keep looking if we're in a Message
+				return true
+			}
+			// keep looking while we're in a proto.Message.
+			_, ok := c.Node().(*proto.Message)
+			return ok
+		},
+		// return immediately iff found.
+		func(*Cursor) bool { return !found })
+	if found {
+		return
+	}
+	return nil, errors.Errorf("field %s not found", name)
```

Comment:
Adding "proto" or similar to the error might add some useful context for the user:
```suggestion
	return nil, errors.Errorf("proto field %s not found", name)
```

Reply:
It would apply also to the other proto field errors down the line in the PR